drivers/mcp2515/mcp2515_spi.h

```diff
@@ -0,0 +1,169 @@
+
+/**
+ * @brief Reset MCP2515 device though SPI interface
+ *
+ * The MCP2515 device is reset by sending the right message over SPI interface.
+ *
+ * @param[in]  dev          device descriptor
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int mcp2515_spi_reset(const candev_mcp2515_t *dev);
+
+/**
+ * @brief Read the register value corresponding to @p addr
+ *
+ * @param[in]  dev          device descriptor
+ * @param[in]  addr         register addr
+ * @param[out] buf          buffer to receive register value
+ * @param[in]  len          length of register value
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int mcp2515_spi_read(const candev_mcp2515_t *dev, uint8_t addr, uint8_t *buf,
+                     unsigned int len);
+
+/**
+ * @brief Read the can data received in the rx @p mailbox
+ *
+ * @param[in]  dev          device descriptor
+ * @param[in]  mailbox      rx mailbox
+ * @param[out] buf          buffer to receive can data
+ * @param[in]  len          length of can data
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int mcp2515_spi_read_rxbuf(const candev_mcp2515_t *dev, uint8_t mailbox,
+                           void *buf, uint8_t len);
+
+/**
+ * @brief Send the register value corresponding to @p addr
+ *
+ * @param[in]  dev          device descriptor
+ * @param[in]  addr         address to write
+ * @param[in]  buf          buffer containing register data
+ * @param[in]  len          buffer length
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int mcp2515_spi_write(const candev_mcp2515_t *dev, uint8_t addr, uint8_t *buf,
+                      unsigned int len);
+
+/**
+ * @brief Send the can data to the tx @p mailbox
+ *
+ * @param[in]  dev          device descriptor
+ * @param[in]  mailbox      tx mailbox
+ * @param[in]  buf          buffer containing can data
+ * @param[in]  len          buffer length
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int mcp2515_spi_write_txbuf(const candev_mcp2515_t *dev, uint8_t mailbox,
+                            void *buf, uint8_t len);
+
+/**
+ * @brief Initiate message transmission.
+ *
+ * The RTS command can be used to initiate message transmission for one or more
+ * of the transmit buffers.
+ *
+ * @param[in]  dev          device descriptor
+ * @param[in]  mailbox      mailbox to enable transmission
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int mcp2515_spi_rts(const candev_mcp2515_t *dev, uint8_t mailbox);
+
+/**
+ * @brief Read MCP2515 status over SPI interface.
+ *
+ * @param[in]  dev          device descriptor
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+uint8_t mcp2515_spi_read_status(const candev_mcp2515_t *dev);
```

Comment:
If ti can return a `<0` then the return should be int

Reply:
`status` is a byte read through `spi_transfer_byte`. According to the [docs](https://doc.riot-os.org/group__drivers__periph__spi.html#ga6e2f58f7b719d1e8ac5cb05bdeed7c9b) that function returns a uint8_t?

Reply:
The doc was wrong in this case. `status` is read through `spi_read_byte`, which returns a uint8_t.